pkg/edition/java/lite/forward.go

```diff
@@ -156,28 +167,104 @@ func findRoute(
 	}
 
 	tryBackends := route.Backend.Copy()
-	nextBackend = func() (string, logr.Logger, bool) {
+	nextBackend = func() (string, bool) {
+		switch route.Strategy {
+		case "random":
+			return randomNextBackend(tryBackends)()
+		case "round-robin":
+			return roundRobinNextBackend(host, tryBackends)()
+		case "least-connections":
+			return leastConnectionsNextBackend(tryBackends)()
+		case "lowest-latency":
+			return lowestLatencyNextBackend(tryBackends)()
+		default:
+			// Default to random strategy
+			return randomNextBackend(tryBackends)()
+		}
+	}
+
+	return log, src, route, nextBackend, nil
+}
+
+func randomNextBackend(tryBackends []string) nextBackendFunc {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	return func() (string, bool) {
 		if len(tryBackends) == 0 {
-			return "", log, false
+			return "", false
 		}
-		// Pop first backend
-		backend := tryBackends[0]
-		tryBackends = tryBackends[1:]
+		randIndex := r.Intn(len(tryBackends))
+		return tryBackends[randIndex], true
+	}
+}
 
-		dstAddr, err := netutil.Parse(backend, src.RemoteAddr().Network())
-		if err != nil {
-			log.Info("failed to parse backend address", "wrongBackendAddr", backend, "error", err)
-			return "", log, false
+var roundRobinIndex = make(map[string]int)
+
+func roundRobinNextBackend(routeHost string, tryBackends []string) nextBackendFunc {
+	return func() (string, bool) {
+		if len(tryBackends) == 0 {
+			return "", false
+		}
+		index := roundRobinIndex[routeHost] % len(tryBackends)
+		backend := tryBackends[index]
+		roundRobinIndex[routeHost]++
+		return backend, true
+	}
+}
+
+var connectionCounts = make(map[string]int)
+var connectionCountsMutex sync.Mutex
```

Comment:
It's bad practice to use global state like this in Go.
There are possibly multiple Gate instances sharing this, which needs to be avoided.